storm-server/src/main/java/org/apache/storm/daemon/supervisor/Slot.java

```diff
@@ -89,54 +94,56 @@ public Slot(AsyncLocalizer localizer, Map<String, Object> conf,
+                containerLauncher,
+                host,
+                port,
+                iSupervisor,
+                localState,
+                this,
+                metricsExec, metricsProcessor);
+
         LocalAssignment currentAssignment = null;
+        Container container = null;
+        LocalAssignment newAssignment = null;
+
+        Map<Integer, LocalAssignment> assignments = localState.getLocalAssignmentsMap();
         if (assignments != null) {
             currentAssignment = assignments.get(port);
-        }
-        Container container = null;
-        if (currentAssignment != null) {
-            try {
-                // For now we do not make a transaction when removing a topology assignment from local, an overdue
-                // assignment may be left on local disk.
-                // So we should check if the local disk assignment is valid when initializing:
-                // if topology files does not exist, the worker[possibly alive] will be reassigned if it is timed-out;
-                // if topology files exist but the topology id is invalid, just let Supervisor make a sync;
-                // if topology files exist and topology files is valid, recover the container.
-                if (ClientSupervisorUtils.doRequiredTopoFilesExist(conf, currentAssignment.get_topology_id())) {
-                    container = containerLauncher.recoverContainer(port, currentAssignment, localState);
-                } else {
-                    // Make the assignment null to let slot clean up the disk assignment.
+            if (currentAssignment != null) {
+                try {
+                    // For now we do not make a transaction when removing a topology assignment from local, an overdue
+                    // assignment may be left on local disk.
+                    // So we should check if the local disk assignment is valid when initializing:
+                    // if topology files does not exist, the worker[possibly alive] will be reassigned if it is timed-out;
+                    // if topology files exist but the topology id is invalid, just let Supervisor make a sync;
+                    // if topology files exist and topology files is valid, recover the container.
+                    if (ClientSupervisorUtils.doRequiredTopoFilesExist(conf, currentAssignment.get_topology_id())) {
+                        container = containerLauncher.recoverContainer(port, currentAssignment, localState);
+                    } else {
+                        // Make the assignment null to let slot clean up the disk assignment.
+                        currentAssignment = null;
+                    }
+                } catch (ContainerRecoveryException e) {
+                    //We could not recover container will be null.
+                }
+
+                newAssignment = currentAssignment;
+                if (/* currentAssignment != null && */container == null) {
```

Comment:
nit: can we remove the comment from in here.  I think it detracts from the readability.